core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -120,16 +172,18 @@ public void deprecated(String msg, Object... params) {
      * @param params The parameters used to fill in the message, if any exist.
      */
     @SuppressLoggerChecks(reason = "safely delegates to logger")
-    void deprecated(Set<ThreadContext> threadContexts, String message, Object... params) {
-        Iterator<ThreadContext> iterator = threadContexts.iterator();
+    void deprecated(final Set<ThreadContext> threadContexts, final String message, final Object... params) {
+        final Iterator<ThreadContext> iterator = threadContexts.iterator();
 
         if (iterator.hasNext()) {
             final String formattedMessage = LoggerMessageFormat.format(message, params);
-
+            final String warningValue = formatWarning(formattedMessage);
+            assert WARNING_HEADER_PATTERN.matcher(warningValue).matches();
```

Comment:
should we extractWarningValueFromWarningHeader to see we got formattedMessage back?